wpcom-legacy-redirector.php

```diff
@@ -149,10 +154,19 @@ static function get_redirect_post_id( $url ) {
 
 		$url_hash = self::get_url_hash( $url );
 
-		$redirect_post_id = $wpdb->get_var( $wpdb->prepare( "SELECT ID FROM $wpdb->posts WHERE post_type = %s AND post_name = %s LIMIT 1", self::POST_TYPE, $url_hash ) );
+		// Allow plugins to disable lowercase.
+		if ( apply_filters( 'wpcom_legacy_redirector_check_lowercase', true ) ) {
```

Comment:
Because "/MyUrl" and "/myurl" is not the same URL. You will need to add a redirect for each version of a “from” URL that you want to redirect.